### PR TITLE
[APS-67] Implement AP Delius Context API offender details data source

### PR DESCRIPTION
**Pull request by danlivings-dxw**
> See [APS-67 on the Approved Premises Jira board](https://dsdmoj.atlassian.net/browse/APS-67).

This PR provides the ability to use the AP Delius Context API as a data source for offender details. This is achieved by implementing the `ApDeliusContextApiOffenderDetailsDataSource` class.

The `ApDeliusContextApiOffenderDetailsDataSource` class performs transformations of the data in order to match the same data structure used by the Community API for backwards compatibility. These are enabled by various utility methods in the `OffenderDetailsUtils.kt` and `BaseHMPPSClient.kt` files.

This PR does not change the data source for offender details as this is down to the individual configuration for each environment.

### CODE REVIEW

File: src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/client/BaseHMPPSClient.kt
Comment (davidatkinsuk) [inline src/main/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/client/BaseHMPPSClient.kt:111]: i like it!

**Review by davidatkinsuk — APPROVED** [review]
LGTM. Not sure if you want to wait on a review from someone a bit more familiarity with the data models. I did check the mapping is 'inverse' to the existing mapping, and looks good to me.

**Review by pezholio — APPROVED** [review]
LGTM - some neat conversion stuff going on here!

**Pull request merged**
